soroban-env-host/src/host.rs

```diff
@@ -1925,4 +1927,46 @@ impl CheckedEnv for Host {
             .with_ledger_info(|li| self.add_host_object(li.network_id.clone()))?
             .into())
     }
+
+    fn get_current_call_stack(&self) -> Result<Object, HostError> {
+        let frames = self.0.context.borrow();
+
+        let get_host_val_tuple =
+            |id: &Hash, function: &Symbol| -> Result<(HostVal, HostVal), HostError> {
+                let id_val =
+                    self.associate_raw_val(self.add_host_object(<Vec<u8>>::from(id.0))?.into());
+
+                let function_val = self.associate_raw_val(function.clone().into());
+                Ok((id_val, function_val))
+            };
+
+        let mut res = HostVec::new(self.0.budget.clone())?;
+        for frame in frames.iter() {
+            let mut inner = HostVec::new(self.0.budget.clone())?;
+
+            match frame {
+                #[cfg(feature = "vm")]
+                Frame::ContractVM(vm, function) => {
+                    let vals = get_host_val_tuple(&vm.contract_id, &function)?;
+                    inner.push_back(vals.0)?;
+                    inner.push_back(vals.1)?;
```

Comment:
It's better to construct a Vector from these elements and then call `HostVec::from_vec(my_vec)`, so that less intermediate objects are created (happens each time calling `push_back`), and less expensive.

Reply:
Same for the outer Vector

Reply:
The intermediate vector operations wouldn't factor into cost then right? Does that matter?

Reply:
Right, it's not factored into cost right now, and I'll need to update the Cost of `HostVec::from_vec` to account for the externally constructed vector size. But this is okay right now, the metering part still has a lot refining to do.

Reply:
Updated

Reply:
Thanks for making the change. Feel free to resolve the comment.